core/api/service/chain/impl/chain_api_impl.cpp

```diff
@@ -71,6 +82,21 @@ namespace kagome::api {
     return results;
   }
 
+  outcome::result<primitives::BlockData> ChainApiImpl::getBlock(
+      std::string_view hash) {
+    OUTCOME_TRY(h, primitives::BlockHash::fromHexWithPrefix(hash));
+    OUTCOME_TRY(block, block_storage_->getBlockData(h));
+    if (block) return block.value();
+    return Error::BLOCK_NOT_FOUND;
+  }
+
+  outcome::result<primitives::BlockData> ChainApiImpl::getBlock() {
```

Comment:
why not getLastFinalizedBlock()?

Reply:
implementation does look like code duplication of ChainApiImpl::getBlock(std::string_view)

Reply:
Not getLastFinalizedBlock because it's an implementation of getBlock RPC method. It does duplicate the other overload, I'll fix it.

Reply:
Ah, wait, it's not that trivial to do, because the other overload receives a hash as a string view, but from block tree we obtain the binary form already. If I use getBlock(std::string_view), I will have to convert the block hash to string first, and then convert it from string back into binary. Not pretty. This could be solved by introducing a private method which takes a block hash in binary, but doing this to avoid duplication of two lines of code seemed too much for me.